### DISCUSSION TITLE
Github workflow unexpected duplicate trigger

### DISCUSSION THREAD

**Discussion by nickhodaly** [community/community#26681] [Actions]
<p>We are using the following event/types to trigger a workflow and it has led to some unexpected behavior for us, specifically when adding a label before opening a pull request.</p>
<pre><code class="lang-auto">on:
  pull_request:
    types: [labeled, opened, synchronize, reopened]
    branches: '*'
</code></pre>
<p>Our intention was to have this workflow be triggered once, when you open/reopen a new PR, push new commits to an open PR, or label a PR. It looks like it is triggering for all of these cases. However, one flow that is causing the workflow to be triggered twice is when someone adds a label before clicking the <code>create pull request</code> button. We see the workflow trigger twice where one is shown as <code>labeled by</code> and another immediately following showing <code>opened by</code>. Is this the expected behavior? Is there something we can do to prevent the workflow from running twice in this case?</p>

**Comment by ViacheslavKudinov** [comment, marked as answer]
<p>Hi <a class="mention" href="/nickhodaly">@nickhodaly</a><br>
I think it is expected as there were several events generated by change of PR.<br>
To fix this we use <code>concurrency</code>. Please, take a look <a href="https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow" class="inline-onebox" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a>.<br>
It’s not exactly what you need, but it can cancel previous run by another event for same PR and leave the last triggered only to finish.</p>

**Comment by nickhodaly** [comment]
<p><a class="mention" href="/viacheslavkudinov">@ViacheslavKudinov</a> Awesome! this helps. Did not know GitHub actions had this capability. :+1:</p>
